crates/iota-macros/src/lib.rs

```diff
@@ -413,7 +413,6 @@ mod test {
     #[test]
     #[should_panic]
     fn test_macro_overflow() {
-        #[allow(arithmetic_overflow)]
```

Comment:
How come this can be removed? There is clearly an overflow?

Reply:
I suppose it's because it's not running clippy with the test target

Reply:
That would be very odd, can you do a little search on this please?

Reply:
So as far as I can tell, this lint is only checked in `--release` mode, and this is under a `#[cfg(test)]` directive, so this cannot ever actually overflow.

Reply:
It does overflow, the test would fail otherwise

Reply:
Oh, that's right, I had it backwards. Then, I really don't know why this doesn't complain, but it doesn't. 🤷‍♀️

Reply:
I'd still like to know what's happening here 🤔

Reply:
`arithmetic_overflow` actually seems to not even exist (anymore)?